pathiomock.go

```diff
@@ -0,0 +1,59 @@
+package pathio
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"sync"
+)
+
+// MockClient mocks out an S3 bucket
+type MockClient struct {
```

Comment:
Have you considered using [testify mock] (https://godoc.org/github.com/stretchr/testify/mock)? It seems like it might be easier to test reading in isolation this way, since you could specify the return value without having to write or directly set the Filesystem. It would probably be tougher to deal with writes and then reads, but I'm not sure how common that would be for unit testing.